content/en/blog/news/2022-04-22-k8s-122-tekton.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Kubernetes 1.22 - Breaking change!"
+date: 2022-04-22
+draft: false
+description: >
+  Jenkins X goes to Kubernetes 1.22 
+categories: [blog]
+keywords: [Community, 2022]
+slug: "kubernetes-1.22-tekton"
+aliases: []
+author: Jenkins X maintainers
+---
+
+To allow Jenkins X to support Kubernetes 1.22, we had to update our version of Tekton. This updated version of Tekton contains breaking changes that has consequences if you made your own custom Jenkins X pipelines.
+
+To make sure that your custom pipelines continue to work after this upgrade, you must edit the resource settings in your pipelines. Otherwise your pipelines will most likely not be able to start at all, or if they do, consume a lot of resources.
+
+
+#### Changes in Tekton version 28
+Tekton made changes in how to calculate the resources needed to run a pipeline, in order to support the concept of [LimitRange](https://kubernetes.io/docs/concepts/policy/limit-range/) in Kubernetes (introduced in Kubernetes version 1.10). Previously, Tekton simply used the maximum requested cpu and memory of any single step, and set that as limits for the all steps in the pipeline. 
+
+[Tekton documentation on how resources are calculated with LimitRange](https://tekton.dev/vault/pipelines-main/limitrange/)
+
+
+#### How to prepare for upgrade
+
+The Jenkins X pipeline catalog has already been upgraded to handle the changes. You can study [this PR](https://github.com/jenkins-x/jx3-pipeline-catalog/pull/984/files) to see how the new resource settings are handled there.
```

Comment:
So we are going to merge the catalog PR before this blog goes live?

Reply:
I guessed that these changes would work in the old version too. I don't really know if that is true yet.

Reply:
It worked on my system. But I suppose the consequences of introducing these changes in the old version are uncertain. I will change the text.

Reply:
> But I suppose the consequences of introducing these changes in the old version are uncertain

These changes are backwards compatible afaict